File: tests/Domain/Identity/UserIdentityTest.php
<?php

namespace Tests\Domain\Identity;

use App\Domain\Identity\UserConnected;
use App\Domain\Identity\UserIdentity;
use App\Domain\Identity\UserRegistered;
use Tests\Domain\FakeEventPublisher;

class UserIdentityTest extends \PHPUnit_Framework_TestCase {
    public function testWhenRegisterUser_ThenUserRegisteredIsRaised() {
        $fakeEventPublisher = new FakeEventPublisher();
        $userId = 'clem@mix-it.fr';

        UserIdentity::register($fakeEventPublisher, $userId);

        \Assert\that($fakeEventPublisher->events)->count(1);
        /** @var UserRegistered $userRegistered */
        $userRegistered = $fakeEventPublisher->events[0];
        \Assert\that($userRegistered->getUserId())->eq($userId);
    }

    public function testWhenLogIn_ThenUserConnectedIsRaised() {
        $fakeEventPublisher = new FakeEventPublisher();
        $userId = 'clem@mix-it.fr';
        $userIdentity = new UserIdentity(array(new UserRegistered($userId)));

        $userIdentity->logIn($fakeEventPublisher);

        \Assert\that($fakeEventPublisher->events)->count(1);
        /** @var UserConnected $userConnected */
        $userConnected = $fakeEventPublisher->events[0];
        \Assert\that($userConnected)->isInstanceOf('App\Domain\Identity\UserConnected');
        \Assert\that($userConnected->getUserId())->eq($userId);
        \Assert\that($userConnected->getSessionId())->notNull();
        \Assert\that($userConnected->getConnectedAt()->diff(new \DateTime())->s)->range(-1, 1);
    }
}

File: Mixter.Domain.Tests/Identity/UserIdentityTest.cs
using System;
using System.Linq;
using Mixter.Domain.Identity;
using Mixter.Domain.Identity.Events;
using NFluent;
using Xunit;

namespace Mixter.Domain.Tests.Identity
{
    public class UserIdentityTest
    {
        private static readonly UserId UserId = new UserId("user@mixit.fr");

        private readonly EventPublisherFake _eventPublisher;
        
        public UserIdentityTest()
        {
            _eventPublisher = new EventPublisherFake();
        }

        [Fact]
        public void WhenRegisterThenRaiseUserRegisteredEvent()
        {
            UserIdentity.Register(_eventPublisher, UserId);

            Check.That(_eventPublisher.Events).Contains(new UserRegistered(UserId));
        }

        [Fact]
        public void GivenUserRegisteredWhenLogThenRaiseUserConnectedEvent()
        {
            var userIdentity = new UserIdentity(new UserRegistered(UserId));

            userIdentity.LogIn(_eventPublisher);

            var evt = _eventPublisher.Events.OfType<UserConnected>().First();
            Check.That(evt.UserId).IsEqualTo(UserId);
            Check.That(evt.ConnectedAt).IsBeforeOrEqualTo(DateTime.Now).And.IsAfterOrEqualTo(DateTime.Now.AddSeconds(-1));
        }
    }
}
